turbopack/crates/turbo-tasks-backend/src/backend/operation/mod.rs

```diff
@@ -656,49 +560,15 @@ impl<B: BackingStorage> TaskGuard for TaskGuardImpl<'_, B> {
         F: for<'a> FnMut(CachedDataItemKey, CachedDataItemValueRef<'a>) -> bool + 'l,
     {
         self.check_access(ty.category());
-        if !self.backend.should_persist() || self.task_id.is_transient() {
-            return Either::Left(self.task.extract_if(ty, f));
+        if !self.task_id.is_transient() {
```

Comment:
All the other functions got a `if !self.task_id.is_transient() && key.is_persistent() {` guard, is this one different intentionally?

Reply:
Yeah, we should add a ty.is_persistent()